### PR TITLE
esbuild: prioritize "module" over "browser"

### PR DESCRIPTION
**Pull request by samouri**
**summary**
Partial for https://github.com/ampproject/amphtml/issues/36453

By default esbuild will utilize browser, module, main in that order ([docs]()). This PR swaps the priorities of module/browser for better tree shaking of our dependencies. In particular, this allows for the elision of the [cache-url](https://github.com/ampproject/amp-toolbox/tree/main/packages/cache-url) package.

**size diff**
```diff
- ✔ dist/v0/amp-story-0.1.mjs               73.40 kb   –          353.52 kb  
- ✔ dist/v0/amp-story-1.0.mjs               73.40 kb   –          353.52 kb 
+ ✔ dist/v0/amp-story-0.1.mjs               71.09 kb   –          347.51 kb  
+ ✔ dist/v0/amp-story-1.0.mjs               71.09 kb   –          347.51 kb  
```

### CODE REVIEW

File: build-system/tasks/helpers.js [hunk 1]
```diff
@@ -456,6 +456,7 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
         incremental: !!options.watch,
         logLevel: 'silent',
         external: options.externalDependencies,
+        mainFields: ['module', 'browser', 'main'],
```
Comment (jridgewell) [inline build-system/tasks/helpers.js:459]: Let's update the test running with this, too.

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: build-system/tasks/helpers.js (hunk 1 above)
Comment (samouri) [reply build-system/tasks/helpers.js:459]: What test?

Comment (jridgewell) [reply build-system/tasks/helpers.js:459]: All of them: https://github.com/ampproject/amphtml/blob/31cf6fd29a797479d9fd22d5eb2fcdd96960797a/build-system/tasks/runtime-test/runtime-test-base.js#L232-L240

Comment (samouri) [reply build-system/tasks/helpers.js:459]: Ah, forgot about this. Sure

**Review by jridgewell — APPROVED** [review]

**Pull request merged**